packages/pwa-buildpack/src/WebpackTools/plugins/UpwardPlugin.js

```diff
@@ -10,13 +10,24 @@ const httpsAgent = new https.Agent({ rejectUnauthorized: false });
 
 class UpwardPlugin {
     constructor(devServer, env, upwardPath) {
-        this.env = env;
+        this.env = Object.assign(
+            {
+                NODE_ENV: 'development'
+            },
+            env
+        );
         this.upwardPath = upwardPath;
-        // Compose `after` function if something else has defined it.
+        // Compose `after` and `before` functions if something else has defined
+        // them.
+        const oldBefore = devServer.before;
+        devServer.before = (app, ...rest) => {
+            app.use(upward.bestPractices());
+            if (oldBefore) oldBefore(app, ...rest);
+        };
         const oldAfter = devServer.after;
         devServer.after = (app, ...rest) => {
-            app.use((req, res, next) => this.handleRequest(req, res, next));
             if (oldAfter) oldAfter(app, ...rest);
+            app.use((req, res, next) => this.handleRequest(req, res, next));
```

Comment:
Why did you move this line after 29?

Reply:
@sirugh PWADevServer now exposes a builtin image resizing middleware at `/img` (though that route is configurable with an env var). Venia is designed to run on PWADevServer in development mode. If PWADevServer is absent, then the route falls to UPWARD to resolve and then the addition to `venia-upward.yml` handles it (by forwarding it to Fastly if possible).

Reply:
Gotcha. Because line 27 will be falsy in non dev.